tests/meson.build

```diff
@@ -27,6 +27,28 @@ write_fuzzer_exe = executable('fuzz_write_repro',
 
 png_dep = dependency('libpng', version : '>=1.6.0', fallback : ['libpng', 'png_dep'])
 
+can_apply_patch = meson.version().version_compare('>= 0.63.0') # Older versions cannot apply the APNG patch
+apng_note = 'Tests require libpng with Animated PNG (APNG) support, see build documentation'
+meson_too_old_msg = 'Meson version is too old to use libpng fallback'
+
+if png_dep.type_name() != 'internal' # compiler.has_function() does not work with wraps
+    if not cc.has_function('png_get_acTL', prefix : '#include <png.h>', dependencies : png_dep)
+
+        warning(apng_note)
+
+        if can_apply_patch == false
+            error(meson_too_old_msg)
+        else
+            warning('Local version of libpng does not support APNG, attempting to use fallback option...')
+            png_dep = subproject('libpng').get_variable('png_dep') # May fail depending on wrap_mode, hence the warning above
+        endif
+    endif
+else # Using fallback
+    if can_apply_patch == false
+        warning(meson_too_old_msg)
```

Comment:
Shouldn't this be an error?

Reply:
So, which should it be, warning or error?

Keen to see this PR merged